nikola/post.py

```diff
@@ -161,6 +166,14 @@ def __init__(
             for lang in sorted(self.translated_to):
                 default_metadata.update(self.meta[lang])
 
+        # Compose paths
```

Comment:
Use an if/else here, perhaps a regular (simpler) for loop instead of convoluted comprehensions with partial items (re-using the first `self.folders` is just a waste of memory and time)

For example:

```python
if self.folder_base is not None:
    # Use translatable destination folders
    self.folders = {}
    for lang in self.config['TRANSLATIONS'].keys():
        self.folders[lang] = os.path.normpath(os.path.join(
            self.folder_base(lang), self.folder_relative))
else:
    # Old behavior (non-translatable destination path, normalized by scanner)
    self.folders = {lang: self.folder_relative for lang in self.config['TRANSLATIONS'].keys()}